lib/Interpreter/IncrementalParser.cpp

```diff
@@ -182,22 +192,32 @@ namespace cling {
     initializeVirtualFile();
   }
 
-  void
+  bool
   IncrementalParser::Initialize(llvm::SmallVectorImpl<ParseResultTransaction>&
                                 result, bool isChildInterpreter) {
     m_TransactionPool.reset(new TransactionPool(getCI()->getSema()));
+    if (!m_TransactionPool) {
+      llvm::errs() << "TransactionPool could not be created.\n";
+      return false;
+    }
+
     if (hasCodeGenerator()) {
       getCodeGenerator()->Initialize(getCI()->getASTContext());
       m_BackendPasses.reset(new BackendPasses(getCI()->getCodeGenOpts(),
                                               getCI()->getTargetOpts(),
                                               getCI()->getLangOpts()));
+      if (!m_BackendPasses) {
```

Comment:
Let's turn these into asserts. If we cannot create `BackendPasses` and `m_TransactionPool` we are doomed anyway.